### ISSUE TITLE
Completed days not shown as completed

### ISSUE THREAD

**Issue by sm-g** [iSoron/uhabits#269]
android 5.0.1, zenfone  z00d

1. Given habit with daily reminder, notification appears
2. After some time (from seconds to few hours) I mark it as "completed" from notification
3. Sometimes occasionally I notice then (next day usually) that that day not marked as completed. In both main screen and calendar view
4. Today something strange occurred, I open calendar editor, and mark _today_ as done. And magically yesterday become marked too, as it should be. Not sure, if the application remembered that day was marked or something other. (Previously I always first mark missed days first.)

**Comment by devurandom** [comment]
I suffer under the same issue.

For me, it also happens that the days where I am 100% sure I succeeded, and that I got a notification and I checked in as successful on it, often show up as unsuccessful in the four-days-history list at the top. When I then tap the X of the day that I succeeded, but that is shown as if I did not, it is still marked with an X / unsuccessful. Only if I tap it a second time, it is marked with a checkmark.

I would suspect, that the day is set to an invalid/garbage value by the reaction to the notification. I was thinking that was maybe, because too much time passed between the notification being triggered and my reacting to it with success/dismiss: it most often seems to happen with the notifications that I sometimes leave open for a whole day until I react - so I thought that maybe the trigger day and the reaction day were mismatching.
